### ISSUE TITLE
export CARGO_HOME inside working directory

### ISSUE THREAD

**Issue by thsioutas** [xd009642/tarpaulin#477]
I'm working on a test where I create an http test server using the "httptest" crate and I create blocking requests to it using "reqwest". The test passes successfully when Cargo home is located in its default location. However, when I alter Cargo home's location by setting the CARGO_HOME environmental variable to a directory inside the working directory (the cargo directory where the source is also located) the reqwest request fails with a timeout and the test fails. "cargo test" works fine btw in both cases.

**Comment by xd009642** [comment]
So I tried setting `CARGO_HOME` on another project and running tarpaulin and it worked without any issue. Could you provide a link to a repo I can reproduce this with?

**Comment by thsioutas** [comment]
Thanks for trying it. I have a created a reprex here: https://github.com/thsioutas/HttptestAndReqwest

**Comment by xd009642** [comment]
Awesome, thanks for that I can now reproduce it on my machine :+1:  I'll update you when I've made some progress

**Comment by xd009642** [comment]
Okay I've cracked this, the CARGO_HOME variable means source code is downloaded into the folder that shows up in the project directory and in the debug info so tarpaulin identifies it as source to cover. Then with tons of extra traces this likely slows down the test to the point where reqwest times out.

I've added home filtering to source analysis and my cargo handling. I just need to add it to the binary instrumentation which is going to be a bit fiddlier, but hopefully I'll have something today.
